sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegenExec.scala

```diff
@@ -206,6 +206,18 @@ trait CodegenSupport extends SparkPlan {
   def doConsume(ctx: CodegenContext, input: Seq[ExprCode], row: ExprCode): String = {
     throw new UnsupportedOperationException
   }
+
+  /**
+   * for optimization to suppress shouldStop() in a loop of WholeStageCodegen
+   *
+   * isShouldStopRequired: require to insert shouldStop() into the loop if true
+   */
+  def isShouldStopRequired: Boolean = {
+    return shouldStopRequired && !(this.parent != null && !this.parent.isShouldStopRequired)
```

Comment:
Is this better to understand?

    def isShouldStopRequired: Boolean = {
      assert(this.parent != null)
      shouldStopRequired && this.parent.isShouldStopRequired
    }